src/webgl/transform-feedback.js

```diff
@@ -23,106 +17,182 @@ export default class TranformFeedback extends Resource {
   constructor(gl, opts = {}) {
     assertWebGL2Context(gl);
     super(gl, opts);
+
+    this.configuration = null;
     this.buffers = {};
+    this.unused = {};
+    this._bound = false;
+
     Object.seal(this);
 
     this.initialize(opts);
   }
 
-  initialize({buffers = {}, varyingMap = {}}) {
-    this.bindBuffers(buffers, {clear: true, varyingMap});
+  initialize(props) {
+    this.configuration = props.configuration || (props.program && props.program.getConfiguration());
+    this.reset();
+    this.setProps(props);
   }
 
-  bindBuffers(buffers = {}, {clear = false, varyingMap = {}} = {}) {
-    if (clear) {
-      this._unbindBuffers();
-      this.buffers = {};
+  setProps(props) {
+    if ('buffers' in props) {
+      this.setBuffers(props.buffers);
     }
-    for (const bufferName in buffers) {
-      const buffer = buffers[bufferName];
-      const index = Number.isFinite(Number(bufferName)) ?
-        Number(bufferName) : varyingMap[bufferName];
-      assert(Number.isFinite(index));
-      this.buffers[index] = buffer;
+  }
+
+  reset() {
+    this.buffers = {};
+    this.unused = [];
```

Comment:
`this.unused = [];` -> `this.unused = {};`

Reply:
Will address both comments in next PR.